src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -300,7 +300,9 @@ internal static Symbol GetRuntimeMember(NamedTypeSymbol declaringType, ref Membe
             ImmutableArray<TypedConstant> arguments = default(ImmutableArray<TypedConstant>),
             ImmutableArray<KeyValuePair<WellKnownMember, TypedConstant>> namedArguments = default(ImmutableArray<KeyValuePair<WellKnownMember, TypedConstant>>))
         {
-            var ctorSymbol = (MethodSymbol)GetWellKnownTypeMember(constructor);
+            DiagnosticInfo useSiteDiagnostics = null;
+            var ctorSymbol = (MethodSymbol)Binder.GetWellKnownTypeMember(this, constructor, out useSiteDiagnostics, true);
```

Comment:
@cston, the change only affects optional attributes. My understanding is that we just don't synthesize such optional attributes if there is an issue with them. 
That's also the behavior on the VB side.

I will double-check with @AlekseyTs to confirm desired behavior.